modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache

```diff
@@ -122,124 +121,253 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
     else if(type.startsWith("{{prefix}}") && obj.isObject()) {
         // complex type
         QJsonObject jsonObj = obj.toObject();
-        {{prefix}}Object * so = ({{prefix}}Object*)::{{cppNamespace}}::create(type);
+        {{prefix}}Object * so = ({{prefix}}Object*)::{{cppNamespace}}::create(complexType);
         if(so != nullptr) {
             so->fromJsonObject(jsonObj);
             {{prefix}}Object **val = static_cast<{{prefix}}Object**>(value);
-            delete *val;
+            if(*val != nullptr) delete *val;
             *val = so;
         }
     }
     else if(type.startsWith("QList") && QString("").compare(complexType) != 0 && obj.isArray()) {
         // list of values
         if(complexType.startsWith("{{prefix}}")) {
-            QList<{{prefix}}Object*>* output = new QList<{{prefix}}Object*>();
+            auto output = reinterpret_cast<QList<{{prefix}}Object *> **> (value);
```

Comment:
Why use auto here?

Reply:
IMO using `auto` is a much better alternative to specifying the full type here, which would be redundant and decrease readability. I don't see any reason for this project to shy away from writing modern C++ code.
@etherealjoy could you please restore the auto keywords?